src/main/java/randoop/generation/literaltfidf/LiteralStatistics.java

```diff
@@ -0,0 +1,208 @@
+package randoop.generation.literaltfidf;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import org.plumelib.util.SIList;
+import randoop.sequence.Sequence;
+import randoop.types.Type;
+
+/**
+ * This class stores information about literals used in one "scope" of the SUT, where a scope is a
+ * class, a package, or the entire SUT.
+ *
+ * <p>For each literal (represented as a sequence), it stores the number of uses of the sequence and
+ * the number of classes that contain the sequence. It also stores the number of classes in the
+ * scope. Literals are segregated by their output type.
+ */
+public class LiteralStatistics {
+
+  /**
+   * Per-output-type index: for each {@link Type}, a map from {@link Sequence} (literal producer) to
+   * its {@link LiteralUses} within the scope.
+   */
+  private final Map<Type, Map<Sequence, LiteralUses>> literalUsesByType = new LinkedHashMap<>();
+
+  /** The number of classes in this scope. */
+  private int numClasses = 0;
+
+  /** Creates a new empty LiteralStatistics. */
+  public LiteralStatistics() {}
+
+  /**
+   * Returns the number of classes in the current scope.
+   *
+   * @return the number of classes in the current scope
+   */
+  public int getNumClasses() {
+    return numClasses;
+  }
+
+  /**
+   * Return the {@link LiteralUses} for the given sequence.
+   *
+   * @param seq a sequence
+   * @return the {@link LiteralUses} for the given sequence
+   */
+  private LiteralUses getLiteralUses(Sequence seq) {
+    Type outputType = seq.getLastVariable().getType();
+    Map<Sequence, LiteralUses> typeMap =
+        literalUsesByType.computeIfAbsent(outputType, k -> new LinkedHashMap<>());
+    LiteralUses currentUses = typeMap.computeIfAbsent(seq, k -> new LiteralUses());
```

Comment:
Why does this code create a new entry just for the purpuose of returning an empty `LiteralUses`?  It would be more direct and efficient to return `LiteralUses.empty()` if the key doesn't exist.

I also have a question about whether any key should be absent.  Should this code actually throw an error, rather than return an empty `LiteralUses`, if a key isn't found?